mesonbuild/interpreter.py

```diff
@@ -3657,8 +3657,16 @@ def func_configure_file(self, node, args, kwargs):
         # Install file if requested, we check for the empty string
         # for backwards compatibility. That was the behaviour before
         # 0.45.0 so preserve it.
+        install = kwargs.get('install', None)
+        if not isinstance(install, bool) and install is not None:
+            raise InterpreterException('"install" keyword must be a boolean')
         idir = kwargs.get('install_dir', None)
-        if isinstance(idir, str) and idir:
+        if not isinstance(idir, str) and idir is not None:
+            raise InterpreterException('"install_dir" keyword must be a string')
+        if install and not idir:
+            mlog.warning('Output file', mlog.bold(ofile_rpath, True),
+                         'has \'install : true\', but no valid install_dir is set')
```

Comment:
I think this should be an error, or at least should be an error eventually. Does anyone else have thoughts?

Reply:
Yes that should raise an exception, it won't be able to install without an install_dir.

Reply:
Mh, I was thinking about this, just I didn't want to break current configurations which might have set an install stanza (although didn't work, was accepted), with wrong `idir`.

But if you guys are fine with that, I'll go with an error.

Reply:
I don't think we need a backwards compatibility story for things that weren't advertised.